File: Backend/tshirt-store-backend/src/main/java/com/tshirtstore/service/admin/AdminCustomerService.java
package com.tshirtstore.service.admin;

import com.tshirtstore.dto.admin.CustomerSummaryResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminCustomerService {
    Page<CustomerSummaryResponse> getCustomers(String search, Pageable pageable);
    CustomerSummaryResponse getCustomerById(Long id);
    void toggleCustomerActiveStatus(Long id);
}

File: Frontend/T-Shirt/src/services/adminCustomerService.ts
import { api } from "./api";
import type { Customer } from "../types/adminCustomer";

export const adminCustomerService = {
  getCustomers: async (page = 0, size = 15, search = ""): Promise<{ content: Customer[]; totalElements: number }> => {
    const response = await api.get("/admin/customers", { params: { page, size, search } });
    return response.data;
  },

  toggleCustomerStatus: async (id: number): Promise<void> => {
    await api.patch(`/admin/customers/${id}/toggle-status`);
  },
};
